test/integration/TokenRejectIntegrationTest.js

```diff
@@ -0,0 +1,834 @@
+                    .sign(receiverPrivateKey);
+
+                try {
+                    await (
+                        await tokenRejectTx.execute(env.client)
+                    ).getReceipt(env.client);
+                } catch (err) {
+                    expect(err.message).to.include("TOKEN_IS_PAUSED");
+                }
+            });
+
+            it("when using Fungible Token id when referencing NFTs", async function () {
+                this.timeout(120000);
+
+                // transfer to receiver
+                const nftId = new NftId(tokenId, 1);
+                await (
+                    await new TransferTransaction()
+                        .addNftTransfer(nftId, env.operatorId, receiverId)
+                        .execute(env.client)
+                ).getReceipt(env.client);
+
+                try {
+                    // reject nft using addTokenId
+                    await (
+                        await (
+                            await new TokenRejectTransaction()
+                                .setOwnerId(receiverId)
+                                .addTokenId(tokenId)
+                                .freezeWith(env.client)
+                                .sign(receiverPrivateKey)
+                        ).execute(env.client)
+                    ).getReceipt(env.client);
+                } catch (err) {
+                    expect(err.message).to.include(
+                        "ACCOUNT_AMOUNT_TRANSFERS_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON",
+                    );
+                }
+
+                try {
+                    // reject nft using setTokenIds
+                    await (
+                        await (
+                            await new TokenRejectTransaction()
+                                .setOwnerId(receiverId)
+                                .setTokenIds(new Array(tokenId))
+                                .freezeWith(env.client)
+                                .sign(receiverPrivateKey)
+                        ).execute(env.client)
+                    ).getReceipt(env.client);
+                } catch (err) {
+                    expect(err.message).to.include(
+                        "ACCOUNT_AMOUNT_TRANSFERS_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON",
+                    );
+                }
+            });
+
+            it("when there's a duplicated token reference", async function () {
+                this.timeout(120000);
+
+                const nftId = new NftId(tokenId, 1);
+
+                // transfer nft to receiver
+                await (
+                    await new TransferTransaction()
+                        .addNftTransfer(nftId, env.operatorId, receiverId)
+                        .execute(env.client)
+                ).getReceipt(env.client);
+
+                // reject nft
+                try {
+                    await new TokenRejectTransaction()
+                        .setTokenIds([tokenId, tokenId])
```

Comment:
Because of [this](https://github.com/hashgraph/hedera-sdk-js/pull/2411/files/1547aece28bf5d9e9f022b0f759f953603d84d10#r1694699095) it's passing. It's using the fungible token ids